src/main/java/mods/railcraft/common/util/routing/RoutingLogic.java

```diff
@@ -95,7 +95,8 @@ private EntityMinecart getRoutableCart(EntityMinecart cart) {
             if (cart instanceof INeedsFuel)
                 return cart;
         }
-        return train.getLocomotive();
+        EntityMinecart loco = train.getLocomotive();
+        return loco == null ? cart : loco;
```

Comment:
This will probably result in trains getting cut in half. Only the leading cart should be evaluated.